faq.html

```diff
@@ -161,6 +161,87 @@ <h2 id="outputformat">The output format of ParticleFile</h2>
+    <li>Do not re-add particles periodically via <code class="language-python">repeatdt</code>, e.g. change <code class="language-python">pset = ParticleSet(..., lat=np.random.rand(16,1), repeatdt=timedelta(hours=1))</code> to <code class="language-python">pset = ParticleSet(..., lat=np.random.rand(16,1))</code></li>
+  </ul>
+  Run your simulation again and observe the memory with the tools mentioned above. If your simulation runs through without problems, it is almost certain your real simulation requires too many particles. There is no generic solution at this point for it, so: Get in touch with us at <a href="https://github.com/OceanParcels/parcels/issues">the OceanParcels GitHub page</a> via the issue tracker to find an individual solution.
+  <p></p>
+  If the reduction of particles indeed does not solve your issue and your simulation still crashes, then this is likely due to the Fields consuming more memory than available. Luckily, there are some solutions to this problem.
+  <p></p>
+
+  <h4>Simulations crash on HPC clusters (via job submission systems such as SGE or SLURM), but locally finishes without problems</h4>
+  <p></p>
+  The behaviour of simulations crashing on cluster systems whereas running as-expected on local machines can occur because
+  <ul>
+    <li>Python itself does not return memory to the operation system</li>
+    <li>your (field) data are too large for the memory, but fit comfortably in <i>virtual memory</i> (e.g. swap drive, swap files, page files) =-> job submission systems do not provide access to the <i>swap</i> memory</li>
+    <li>your (field) data exceed the default memory capacity granted to users of your cluster -> consult tutorials on <b>SGE qsub</b> (e.g. from <a href="http://star.mit.edu/cluster/docs/0.92rc2/guides/sge.html">MIT</a> and <a href="https://wikis.utexas.edu/display/CCBB/sge-tutorial">University of Austin / Texas</a>) or <b>SLURM</b> (e.g. from <a href="https://researchcomputing.princeton.edu/education/online-tutorials/getting-started/introducing-slurm">Princeton University</a> or <a href="https://ubccr.freshdesk.com/support/solutions/articles/5000688140-submitting-a-slurm-job-script">University of Buffalo</a>) to raise the granted memory to your simulation job. In particular, consider submission option parameters such as <i>-l h_vmem=<b>size</b></i> (SGE qsub) and <i>
+      --mem=<b>size</b></i> (SLURM).</li>
+  </ul>
+
+  <p></p>
+  <h4>Introduce field chunking (i.e. <i>dynamic loading</i> of fields)</h4>
+  <p></p>
+  So, your situation is as follows: you found out that your Parcels simulation consumes more field data than can fit in memory. This also occurs irrespective of the platform (ie. operation system; system configuration) you run the simulations on - they fail on your workstation, your laptop, and your cluster (via job submission or otherwise). Then, it may be that you are not using field chunking.
```

Comment:
Do we really expect people to test on multiple operating systems?

Reply:
hm, perhaps you can help me formulate that.

What I want to say is: if you work on a cluster, remote desktop, CITRIX workstation environment, *what-not* - test you program also in a simple, controlled environment (if possible). It doesn't matter which operation system is being used, it doesn't matter which general computer is used (unless it's something really special - AIX, mobile phone, embedded system, ...), it is just meant to make sure the user/domain expert eliminates the potential for external error sources, or perhaps find out which of the external sources produces the error.

It's also meant in relation to the passage before: "After making sure it's not the job system from the cluster, still getting the same error, what to check next ?"

Reply:
Ah, then it makes sense. Perhaps simply remove the line

"This also occurs irrespective of the platform (ie. operation system; system configuration) you run the simulations on - they fail on your workstation, your laptop, and your cluster (via job submission or otherwise)."

And if you want you can add
"[... can fit in memory] on any system you tried" to the end of the first line